native/android/SwrveSDKUnityBridge/src/main/java/com/swrve/sdk/SwrveUnityCommon.java

```diff
@@ -0,0 +1,392 @@
+package com.swrve.sdk;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+
+import com.google.gson.Gson;
+import com.google.gson.internal.LinkedTreeMap;
+import com.google.gson.reflect.TypeToken;
+import com.swrve.sdk.conversations.ui.ConversationActivity;
+import com.unity3d.player.UnityPlayer;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.ref.WeakReference;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SwrveUnityCommon implements ISwrveCommon, ISwrveConversationSDK
+{
+    public final static String SWRVE_TEMPORARY_PATH_KEY = "swrveTemporaryPath";
+    public final static String SDK_VERSION_KEY = "sdkVersion";
+    public final static String PREFAB_NAME_KEY = "prefabName";
+    public final static String DEVICE_INFO_KEY = "deviceInfo";
+
+    private final static String LOG_TAG = "UnitySwrveCommon";
+
+    private Map<String, Object> currentDetails;
+    protected WeakReference<Context> context;
+
+    private String sessionKey;
+
+    private File cacheDir;
+
+    private static SwrveUnityCommon instance;
+
+    public static boolean isInitialised() {
+        return null != instance;
+    }
+
+    public SwrveUnityCommon() { }
+
+    public void init(String jsonString) {
+        init(
+            UnityPlayer.currentActivity.getApplicationContext(),
+            jsonString
+        );
+    }
+
+    /***
+     * This is the most proper Constructor to call from the Unity layer,
+     * via a native plugin, with a jsonString of settings, which will be
+     * then cached locally.
+     *
+     * @param context ApplicationContext
+     * @param jsonString JSON String of configuration
+     */
+    public void init(Context context, String jsonString) {
```

Comment:
Everything here should be part of the constructor. Or else else call init from constructor and make this method private.